contracts/src/VectorX.sol

```diff
@@ -145,25 +191,27 @@ contract VectorX is IVectorX, TimelockedUpgradeable {
             500000
         );
         emit HeaderRangeRequested(
-            _trustedBlock,
+            latestBlock,
             trustedHeader,
             _authoritySetId,
             authoritySetHash,
             _requestedBlock
         );
     }
 
-    /// @notice Add target header hash, and data + state commitments for (trustedBlock, targetBlock].
-    /// @param _trustedBlock The block height of the trusted block.
-    /// @param _authoritySetId The authority set id of the header range (trustedBlock, targetBlock].
+    /// @notice Add target header hash, and data + state commitments for (latestBlock, targetBlock].
+    /// @param _authoritySetId The authority set id of the header range (latestBlock, targetBlock].
     /// @param _targetBlock The block height of the target block.
     /// @dev The trusted block and requested block must have the same authority set id.
     function commitHeaderRange(
-        uint32 _trustedBlock,
         uint64 _authoritySetId,
         uint32 _targetBlock
     ) external {
-        bytes32 trustedHeader = blockHeightToHeaderHash[_trustedBlock];
+        if (frozen) {
+            revert ContractFrozen();
+        }
+
+        bytes32 trustedHeader = blockHeightToHeaderHash[latestBlock];
```

Comment:
We should be careful with this change and make sure it runs in prod. In particular, I think there are some weird griefing vectors that might get introduced by this.

Reply:
What are the griefing vectors you're thinking of?